### ISSUE TITLE
How to get in touch regarding a security concern

### ISSUE THREAD

**Issue by psmoros** [statsmodels/statsmodels#9232]
Hello 👋

I run a security community that finds and fixes vulnerabilities in OSS. A researcher (@tvnnn) has found a potential issue, which I would be eager to share with you.

Could you add a `SECURITY.md` file with an e-mail address for me to send further details to? GitHub [recommends](https://docs.github.com/en/code-security/getting-started/adding-a-security-policy-to-your-repository) a security policy to ensure issues are responsibly disclosed, and it would help direct researchers in the future.

Looking forward to hearing from you 👍

(cc @huntr-helper)

**Comment by toolslive** [comment]
Hi, think I found an issue (might be the same as the potential issue above, or a different one).  So can we move forward with this? 

The alternative is just adding an issue with a tag `security` but I would like to give you a head start.

**Comment by bashtage** [comment]
Is it the use of pickle?

**Comment by toolslive** [comment]
No.

**Comment by toolslive** [comment]
the "privately-reporting-a-security-vulnerability" flow doesn't seem to work.
Alternatively I could just post  a proof of concept exploit here. The maintainers had ample opportunity to react.

**Comment by bashtage** [comment]
Just go ahead and post it.

**Comment by josef-pkt** [comment]
or send a private email to me and bashtage

**Comment by josef-pkt** [comment]
> the "privately-reporting-a-security-vulnerability" flow doesn't seem to work.

we have not enabled it yet.
My question was whether we need or should enable it.

In general, statsmodels is intended for interactive use or automated use.
E.g. formula handling by patsy and formulaic use eval. It's the responsibility of the caller not to do anything unsafe.

In statsmodels itself, I think we are not doing anything that can cause security concerns.

**Comment by toolslive** [comment]
indeed:  patsy's eval
An example
```
$> ipython
IN [0]: import statsmodels.formula.api as smf
In [1]: mod = smf.ols(formula="print('cucu')", data={}) 
cucu
...
ValueError: negative dimensions are not allowed
```
By the time you get the exception, it's too late.

>  It's the responsibility of the caller not to do anything unsafe.

I think you need to re-evaluate this. Your average user is a not a programmer but a scientist. In this particular case, I accidentally discovered it after being contacted by a puzzled statistician.

Btw, the code suggests there's a different (safer) way  to define the formula via a generic formula object. However it's utterly unclear how that should be accomplished.

**Comment by bashtage** [comment]
I don't think we are going to address this (or practically can).  We will likely eventually move to formulaic, although this project also used `eval`.
